### DISCUSSION TITLE
Cevopen: annotation of terpene pathway images

### DISCUSSION THREAD

**Discussion by petermr** [petermr/petermr#4] [Ideas]
Subproject to annotate images from Alex Pico.
Relates to https://github.com/petermr/cevopen

# requirements
* goals of project
* column-dictionary . What columns do we need (with brief description)
* data entry protocol
* analysis of results

**Comment by Vaish2611** [comment]
We would be classifying terpene pathway

**Comment by petermr** [comment]
# Raw_Data
raw data is from 
 Alex on https://github.com/wikipathways/wikipathways-development/discussions/33
where we can find out what his tool does.

**Comment by ananyas168** [comment]
Regarding data entry protocol:

1. First is to decide which csv editor to use to fill the columns.
2. Once we have the csv we will merge it will the existing one based on pcmid and other characteristics of each image.

**Comment by ananyas168** [comment]
### Question regarding data entry, result anlysis 
1.a. How are you planning to give images to verifiers for classification?

> Alex has a system for iterating over the images. If it doesn't work for us we'll need to use the current approach I think

 1.b.  Along with the images are you going to give them a csv file?

> Yes. That's what @Vaishnavi Jha will coordinate

1.c. If so what does this CSV contain. Precisely what are the columns this CSV contains?

> we'll` hear her ideas!.


2. Now about the categorical columns...will the verifier have to produce their own csv file with the categorical column and fill yes or no?

> I would expect so

2.b. If so which csv editor they will use?

> there's a list here: https://www.slant.co/topics/15897/~csv-editors        . I haven't used any of them so we'll need to get people's feedback

3. After we get the csv file from each verifier what do you want us to do with it?

> good question. we'll need a Pandas command to compare agreement between annotators.  It could be everyone-by-everyone (i.e. N(N-1)/2 comparisons. There will be tools to help us do this, I'm sure

**Reply by ananyas168** [reply]
> we'll need a Pandas command to compare agreement between annotators. It could be everyone-by-everyone (i.e. N(N-1)/2 comparisons. There will be tools to help us do this, I'm sure

Yes, @petermr, I think it can be done easily. Once we have the data it will be easy for us to visualize it and devise a solution.

**Comment by Vaish2611** [comment]
<html>
<body>
<b style="font-weight:normal;" id="docs-internal-guid-50722006-7fff-0b00-7731-bd5597789b9f"><div dir="ltr" style="margin-left:0pt;" align="left">

PMCID | Fig No. | Pathway | Metabolic | Terpene
-- | -- | -- | -- | --
  |   |   |   |  

</div></b>
</body>
</html>

**Rules for identifying Pathway:**

#https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6394206/figure/fig5/
- Any series of event leading to a product or depiction of chain of events

**Rules for identifying Metabolic Pathway:**

- See if it has biochemical molecules for example as shown in fig, they are generally, carbohydrates, lipids, nucleic acids, and proteins: (refer this image below)
#https://images.app.goo.gl/KyyrSAwK7AhUSUs37
- In general a metabolic pathway are linked series of biochemical reactions
- Key Point: Metabolites that comprises reactant, product and intermediates product of enzymatic reactions.
- Could also see there would be catalysts: enzymes, co-factors etc.

**Rules for identifying Terpene Pathway:**

- Generally anything ending with terpene, Basic unit of terpene is isoprene:
(refer this image below)
#https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.researchgate.net%2Ffigure%2FClassification-of-terpenes-according-to-the-number-of-isoprene-units-41-44_tbl1_253342367&psig=AOvVaw2tXI5I41c2Fo237DQJs-PY&ust=1641911387013000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCODO-OKyp_UCFQAAAAAdAAAAABAD
- Terpenes are alkenes so mostly the compounds ending with “ene” for eq: limonene,myrcene 
- Sometimes they might have functional group like alcohol so compounds ending with “ol” for eg: terpineol, retinol, hinokitiol
- Then, see if somewhere IPP —> DMAPP, this kind of conversion is occurring. (refer this image below)
#https://www.google.com/url?sa=i&url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FIsopentenyl-diphosphate_delta_isomerase&psig=AOvVaw11riGUuKqbTxX43OvNxRft&ust=1641911158420000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCLi-ufSxp_UCFQAAAAAdAAAAABAD
- Now if these elements are present, we just have to spot if there’s a mevalonate or mep pathway, as we could produce terpene only by these two pathway:
---For MVA Pathway, there’s mevalonic acid present (key point), here acetyl Co-A (key point) is being converted through series of enzymatic reaction into IPP —> DMAPP
                 or
---For MEP Pathway, there’s methylerythritol 4-phosphate present (key point), here Pyruvate + glyceraldehyde 3 phosphate (key point) is being converted through series of enzymatic reaction into IPP —> DMAPP

**Comment by Vaish2611** [comment]
This is how one annotators work might look like!
<html>
<body>
<b style="font-weight:normal;" id="docs-internal-guid-654f8eb9-7fff-2696-5cf0-a730212de592"><div dir="ltr" style="margin-left:0pt;" align="left">

PMCID | Fig No. | Pathway | Metabolic | Terpene
-- | -- | -- | -- | --
PMC6323817 | Fig. 6 | Yes | No | Yes
PMC6348685 | Fig. 4 | Yes | Yes | Yes
PMC6356271 | Fig. 2 | Yes | No | No
PMC6360084 | Fig. 8 | Yes | Yes | Yes
PMC6374618 | Fig. 4 | Yes | Yes | Yes
PMC6394206 | Fig. 5 | Yes | No | No
PMC6412517 | Fig. 2 | Yes | Yes | Yes
PMC6423058 | Fig. 2 | Yes | Yes | Yes
PMC6437448 | Fig. 1 | Yes | No | No
PMC6441073 | Fig. 3 | Yes | No | No

</div></b>
</body>
</html>

**Reply by ananyas168** [reply]
The CSV format for the above table can be found [here](https://drive.google.com/file/d/1oaRo79DF5aCHFZbW4iPgxrqrVpjgDqQY/view?usp=sharing)

**Comment by petermr** [comment]
Thanks Vaishnavi,
Looks good,
Is the table available as a CSV?


On Tue, Jan 11, 2022 at 3:54 PM Vaishnavi ***@***.***> wrote:

> This is how one annotators work might look like!
> * PMCID Fig No. Pathway Metabolic Terpene PMC6323817 Fig. 6 Yes No Yes
> PMC6348685 Fig. 4 Yes Yes Yes PMC6356271 Fig. 2 Yes No No PMC6360084 Fig. 8
> Yes Yes Yes PMC6374618 Fig. 4 Yes Yes Yes PMC6394206 Fig. 5 Yes No No
> PMC6412517 Fig. 2 Yes Yes Yes PMC6423058 Fig. 2 Yes Yes Yes PMC6437448 Fig.
> 1 Yes No No PMC6441073 Fig. 3 Yes No No *
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/4#discussioncomment-1948672>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS3FFPKRIREPRAUPEJ3UVRHB5ANCNFSM5LMKFUJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Reply by Vaish2611** [reply]
No peter it isn't.

**Comment by petermr** [comment]
All tables should ideally be in CSV, not XLSX or GoogleDocs.

P.

On Wed, Jan 12, 2022 at 11:58 AM Vaishnavi ***@***.***> wrote:

> No peter it isn't.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/4#discussioncomment-1954323>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS7C26SWZVHPCIB4HOTUVVUGTANCNFSM5LMKFUJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by ananyas168** [comment]
[here](https://drive.google.com/file/d/1oaRo79DF5aCHFZbW4iPgxrqrVpjgDqQY/view?usp=sharing) is the csv file format for the same table shared @Vaish2611
### Pandas DataFrame with hyperlinks
Below is the link to jupyter notebook for creating pandas dataframe with clickable hyperlinks for the demo csv file   [here](https://drive.google.com/file/d/1oaRo79DF5aCHFZbW4iPgxrqrVpjgDqQY/view?usp=sharing) generated by the annotator.
[tutorial on hyperlink pandas dataframe](https://github.com/petermr/CEVOpen/blob/master/creating_clickable_pandas_dataframe.ipynb)

**Comment by ananyas168** [comment]
@petermr  I have requested **Alex Peco** on providing us the complete list of the data along with link for each figure as the CSV file.  

As when we are downloading the list from the website [terepene_pathways](https://gladstone-bioinformatics.shinyapps.io/shiny-terpeneplant/), it doesn't give us the links for the respective entries. Manually adding the link for each figure is very cumbersome and prone to error. 

@Vaish2611 How did you add the link for each figure in the table? Was it manually done?

**Reply by Vaish2611** [reply]
Yes @ananyas168 i added it manually. Sorry didn't saw your message.

**Comment by ananyas168** [comment]
# Issues Faced while the Computing correlation between annotators Data
Below I have described a demo example of annotators data and the issue faced by us while computing correlation coefficient using the Pearson method.
Let's say we have three annotators, Anotater_1, Anotater_2, Anotater_3 and the labels provided by them on 5 different figures (as whether it is a terpene pathway or not). Below is the dataframe containing the data from the three annotators, where each column represents the labels provided by the respective annotators on 5 different figures.

                       Annotator_1   Annotator_2   Annotator_3

        fig_0              yes          yes           yes
        fig_1               no           no           no
        fig_2               no          yes           yes
        fig_3               no           no           yes
        fig_4              yes           no           yes


If we carefully observe the above data, we can see that:

1. **Annotator_1** and **Annotator_2** have the **same opinion** about **fig_0, fig_1, and fig_3** and **different opinions** about  **fig_2, fig_4.** So, they have an **agreement for 3 figures out of 5.** 
2.  **Annotator_2** and **Annotator_3** have the **same opinion** about **fig_0, fig_1, and fig_3** and **different opinions** about  **fig_2, fig_4**. So, they also have an **agreement for 3 figures out of 5.** 
3. Similarly, **Annotator_1** and **Annotator_3** have the **same opinion** about **fig_0, fig_1 and fig_4** and **different opinions** about  **fig_2, fig_3**. so, they have **an agreement for 3 figures out of 5**. 

Technically, since the no of agreement between each annotator boils out to be 3 out of 5, the correlation coefficient should be the same between them. 

But when we use Pandas correlation function, which uses the Pearson method to compute the correlation coefficient, we get the following result:

        Correlation matrix :
                              Annotator_1  Annotator_2  Annotator_3
               Annotator_1     1.000000     0.166667     0.408248
               Annotator_2     0.166667     1.000000     0.408248
               Annotator_3     0.408248     0.408248     1.000000

 
The above correlation matrix tells that
> The correlation coefficient between pair( Annotator_1 and Annotator_2 ) is different from pair (Annotator_2 and Annotator_3) and pair (Annotator_2 and Annotator_3)

Thus we are facing a limitation using the pandas' correlation method. According to our inference of the data, the correlation between the annotators should be the same, but the pandas Pearson correlation gives a different correlation coefficient to the pairs.

With the above limitation, we ought to ask these questions:

1. Is Pearson Correlation a good method to compute the correlation between two categorical columns for the given data? If not what method should we use to compute correlation between categorical columns?
2. Shall we make our own correlation computing function that will satisfy our needs?

It will be great if we can get some help and help. 
Thank you

**Reply by ananyas168** [reply]
### **Python code for the above demo annotation data and correlation coefficient computation is pasted below:**

```
#### import pandas
import pandas as pd

##### initializing a dataframe
data= pd.DataFrame()

##### labels provided by the annotator
anno_1=['yes', 'no', 'no', 'no','yes']
anno_2=['yes', 'no','yes', 'no', 'no']
anno_3=['yes', 'no','yes','yes','yes']

#### adding labels provided by the annotators to the empty pandas dataframe 'data'
data['Annotator_1']= anno_1
data['Annotator_2']= anno_2
data['Annotator_3']= anno_3
#print('data:\n', data)


###### Replacing 'yes' and 'no' in the dataframe with 1 and 0 resoectively for 
######              further computing the correlation coefficient 
data.replace(('yes', 'no'), (1, 0), inplace=True)
#print('data:\n', data)


#### printing  the correlation coefficient computed using Pearson Method.
print('Correlation matrix :\n', data.corr())`
```

**Comment by petermr** [comment]
**Statistical Tests**
https://www.statology.org/cohens-kappa-calculator/
From Chris Hartgerink:
>>You could use something like Cohen's Kappa to get the interrater reliability for two raters https://www.statology.org/cohens-kappa-calculator/
Generalizing that method to three / N-raters is not necessarily too easy..

**Reply by ananyas168** [reply]
Thank you @petermr. I will look into the links and explore the Cohens-kappa-calculator.

**Comment by petermr** [comment]
Thanks,
To do this properly is probably harder than it looks. But for us we have
simple goals:
* how well do the annotators agree? I would give equal weight to Y,Y and N,N
* can we analyse the performance and give a number?
* can we select those Figures where we all agree?


On Tue, Jan 18, 2022 at 1:37 PM Ananya ***@***.***> wrote:

> Thank you @petermr <https://github.com/petermr>. I will look into the
> links and explore the Cohens-kappa-calculator.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/4#discussioncomment-1990586>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS4DYWRC6NJZKTWP3G3UWVUJBANCNFSM5LMKFUJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by AlexanderPico** [comment]
I think I have a really useful tool to help with this project. It might not be clear from the documentation so far, but the tool is easily customized.  Please see the README here and refer to the screenshot (https://github.com/wikipathways/pfocr-curation). 

Now, image this interface (figure information, figure image and classification buttons), but customized for your project. Here are the steps:
 1. Make a filtered RDS database of the ~350 terpene-related pathways. (This would take me ~10 minutes)
 2. Update the buttons per your preference.
   a. Provide me with a list, e.g., MVA and MEP, "pre-terpene" (or whatever you want)
   b. Customize the buttons. (This would take me ~30 minutes)

Then you can run the tool and quickly classify the figures with a GUI!

If this is something you want, just let me know and provide me with (2a) above. If instead you just want a spreadsheet, then let me know and I'll make that instead.

**Reply by petermr** [reply]
That's wonderful!
@Vaish2611 is leading the human annotation sub-project.

@Vaish2611 this is  a great opportunity to investigate the tool and report to this channel.

@AlexanderPico we have a set of rules that @Vaish2611 has coordinated that we are going to test - can someone determine whether something is a TPS pathway from these rules (even if they are not a bisoscientist. Then we'll have a metric (from @ananyas168) of how good the rules are . After that we'll hope to create a "gold standard" of terpene pathways that we can use as input to the image analyzer.

We are also hoping to induct a (software) volunteer who will be working on the text of Figure captions and using NLP.

**Reply by Vaish2611** [reply]
YES @petermr, I will update the rules here after today's meet.

**Comment by Vaish2611** [comment]
With this we confirm that we have a Terpene Pathway. 
1. Only If the following figure is a pathway, can it be a metabolic pathway. 
2. Only If the following figure is a metabolic pathway, can it be a terpene biosynthetic pathway.
3. If any part of the image contains our target pathway we say yes to that entry. 

**Please fill each of these columns with "yes" and "no".**

PMCID | Fig No. | Pathway | Metabolic | Pre-Terpene | Terpene
-- | -- | -- | -- | -- | --
  |   |   |   |   |  

</div><br /></b>

**Rules for identifying Pathway:**
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6323817/figure/Fig6/
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6356271/figure/cells-08-00058-f002/
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6441073/figure/Fig3/

1. Any series of event leading to a product or depiction of chain of events
2. Breakdown of a substrate into various sub products or transfer of metabolites or genetic transformation under a closed boundary of a cell wall or inside a cytoplasm. 
3. Multiple arrows or interconversions between elements can be our indicator. 
   * https://www.genome.gov/sites/default/files/inline-images/BioPathways_Fact-sheet2020.jpg

**Rules for identifying Metabolic Pathway:**

1. See if it has biochemical molecules for example as shown in fig, they are generally, carbohydrates, lipids, nucleic acids, and proteins: 
   * https://images.app.goo.gl/KyyrSAwK7AhUSUs37 Some examples to identify.
2. In general a metabolic pathway are linked series of biochemical reactions
3. Key Point: Metabolites that comprises reactant, product and intermediates product of enzymatic reactions.
4. Could also see there would be catalysts: enzymes, co-factors etc. 
5. If the pathway ends with a molecule not a process then it's a metabolic pathway. 

Examples of metabolic pathway:
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6437448/figure/fig1/
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6453882/figure/Fig4/
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6589726/figure/mpp12802-fig-0001/

**Rules for identifying Pre-Terpene Pathway:**
1. First, look for anything ending with terpene, Basic unit of terpene is isoprene:
   * https://www.researchgate.net/figure/Classification-of-terpenes-according-to-the-number-of-isoprene-units-41-44_tbl1_253342367
2. Terpenes are derived from alkenes so often the compounds ending with “ene” for eq: limonene, myrcene 
3. All terpenes are derived from a small number of starting point (isoprenoids) with multiple double bonds.
4. Many diagrams give an explicit pathway (MVA/MEP) which creates starting terpenes mentioned below:
   * farnesene
   * farnesyl pyrophosphate (FPP)
   * Geranyl Diphosphate (GPP)
   * Geranylgeranyl diphosphate (GGPP)
   * Squalene    
5. They often have added oxygen (eg: alcohols, ketones, acids, asters). For eg: terpineol, retinol, hinokitiol.  
   * https://www.researchgate.net/figure/structures-of-some-common-terpenes_fig1_315777371 These are ome common examples 
6. Then, see if somewhere **IPP —> DMAPP**, this kind of conversion is occurring.  (identifying) 
   * https://en.wikipedia.org/wiki/Isopentenyl-diphosphate_delta_isomerase
   Now if these elements are present, we just have to spot if there’s a mevalonate or mep pathway, as we could produce terpene only by these two pathway:
       * For MVA Pathway, there’s mevalonic acid present (key point), here acetyl Co-A (key point) is being converted through series of enzymatic 
         reaction into IPP —> DMAPP
                                                                         or
       * For MEP Pathway, there’s methylerythritol 4-phosphate present (key point), here Pyruvate + glyceraldehyde 3 phosphate (key point) is being 
         converted through series of enzymatic reaction into IPP —> DMAPP

Example of Pre-Terpene Pathway:
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6360084/figure/fig-8/
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6374618/figure/F4/

**Rules for identifying Terpene Pathway:**

1. It’s a continuum to the pre terpene pathway; a figure could have a diagrammatic representation of terpene conversion without mev/mva listed there; those are listed as terpene pathways. 
2. The ending of the pathway should have unique terpenes, should not be ending with farnesene, farnesyl pyrophosphate (FPP), Geranyl Diphosphate (GPP), Geranylgeranyl diphosphate (GGPP), Squalene.  
3. Also if any common derivatives of terpene i.e. Monoterpenes, Diaterpenes, Sesquiterpenes, Sesterpenes, Triterpenes, Tetraterpenes and Polyterpenes are present as end product don’t mark it as a terpene pathway.  
4. Some common terpene inputs that may be an indicator like: TPS (Terpene Synthase)  which enzyme (catalyst) is responsible for ring formation or addition of a functional group (-OH, =O, acid etc.) to a biosynthetic molecule. 
5. If there’s conversions happening and leading to formation of terpene we could definitely say it’s a terpene pathway.
6. After going through all this rules only one can state that it’s a terpene biosynthetic pathway

Examples of Terpene Pathway:
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6412517/figure/ijms-20-00814-f002/
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6394206/figure/fig5/
* https://www.ncbi.nlm.nih.gov//pmc/articles/PMC6412517/figure/ijms-20-00814-f002/
</body>
</html>

**Reply by anuvc** [reply]
This looks great @Vaish2611 ! At some point we should also add this list to the wiki for a more permanent documentation of such information.

**Reply by Vaish2611** [reply]
Ya definetely @anuvc will surely do that.

**Comment by AlexanderPico** [comment]
Ok. Check out the "Terpene Edition" of the pathway figure curation tool: https://github.com/wikipathways/pfocr-curation-terpene-edition.

I basically added a new section of checkboxes for "Pathway", "Metabolic", "Pre-Terpene", and "Terpene." Follow the rules defined by your team (above) to classify a given pathway figure and then click save. The tool will load one figure at a time and you can quickly classify hundreds of figures this way.

The output of the tool is a TSV file that looks just like the one provided by @Vaish2611 above. I used 1's and 0's to denote whether a particular category applies to a figure or not.  This can easily be customized if you choose to have different classifications. Here is a sample of the output:
![Screen Shot 2022-01-20 at 3 17 57 PM](https://user-images.githubusercontent.com/1418193/150437406-51f66f81-6698-470b-9988-44a08c971328.png)


The tool also outputs an RDS file will all of the curated information, including any fixes you make to the figure number, figure title and caption. These are sometimes incorrect or unclear from the authors and journals. This same tools offers the ability to fix these in our collected database.  If your team chooses to help in curating this information (in addition to terpene classification), then please just share the final RDS files with me and I'll save your fixes.

Final note: The tools starts in a "Training" mode, so you can try it out without worrying about messing anything up. When you are ready to record your curation and classification activity, simply change [this line](https://github.com/wikipathways/pfocr-curation-terpene-edition/blob/master/app.R#L12) to FALSE.

Read the README for more details and a screenshot. Let me know if you have any questions. Hope this is useful for your project!

**Comment by ananyas168** [comment]
### Link to [**Colab Notebook**](https://github.com/petermr/CEVOpen/blob/master/reading_annotator_label_and_correlation_computation.ipynb) which reads the CSV provided by annotators and compare their performance based on correlation score.
---
#### information regarding Cohen's Kappa score and confusion matrix:

Cohen's kappa test is used to examine the reliability between two testers/labelers/annotators:
You can find in-depth details about cohen's kappa by following this [link](https://staff.fnwi.uva.nl/r.fernandezrovira/teaching/MoLProject2011/annotation-reliability.pdf).

Below I have  presented the Cohens kappa calculation on our demo data:

demo data:

               Annotator_1 Annotator_2 Annotator_3 Annotator_4 Annotator_5
     fig_0         yes         yes         yes         yes          no
     fig_1          no          no          no          no         yes
     fig_2          no         yes         yes          no         yes
     fig_3          no          no         yes          no         yes
     fig_4         yes          no         yes         yes          no


Cohens kappa matrix:

                         Annotator_1  Annotator_2  Annotator_3  Annotator_4  Annotator_5
          Annotator_1     1.000000     0.166667     0.285714     1.000000    -0.923077
          Annotator_2     0.166667     1.000000     0.285714     0.166667    -0.153846
          Annotator_3     0.285714     0.285714     1.000000     0.285714    -0.363636
          Annotator_4     1.000000     0.166667     0.285714     1.000000    -0.923077
         Annotator_5     -0.923077    -0.153846    -0.363636    -0.923077     1.000000



Moreover, we have decided to move forward with a confusion matrix to understand what the performance between two annotators looks like:

 confuion matrix between Annotator_1 and Annotator_2:

             yes  no
       yes    1   1
       no     1   2

**Comment by Vaish2611** [comment]
Here's the CSV file to be filled. **Please fill each of these columns with "yes" and "no".**
[Classification.of.30.fig.csv](https://github.com/petermr/petermr/files/7971900/Classification.of.30.fig.csv)
Refer to these [rules](https://github.com/petermr/CEVOpen/wiki/Rules-for-Classification-of-terpene-pathway) classification.

**Reply by anuvc** [reply]
The link to the rules throws `page not found` error. I think you need to update the link.

Consider copying the rules from discussion to a wiki, and providing the wiki link here. That will be better for sharing. Let me know if you need help creating a wikipage.

**Reply by Vaish2611** [reply]
Thanks anuv for telling, I have updated the link with wiki page.

**Reply by petermr** [reply]
If anyone is reading and writing CSV files from Python ALWAYS use a library. Here's a typical 15-min video (there are lots of tutorials) https://www.youtube.com/watch?v=q5uM4VKywbA&ab_channel=CoreySchafer

and here's a tutorial

https://realpython.com/python-csv/

**Reply by Vaish2611** [reply]
Thanks peter, its very helpful.

**Reply by anuvc** [reply]
Looks great! 👍

**Comment by roopavsm** [comment]
Here's the link containing the code to convert .rds files to pandas
https://github.com/roopavsm/codework/blob/554fdd4afdea0ac47b50b73210e21efae7d11c20/Converting_rds_to_pandas.ipynb

**Comment by petermr** [comment]
# Daily meetings:

(add following meetings here)
(sorry Github drops the list buttons)

## TUESDAY 2022-01-25
Meeting at 1830 IST 1300 UTC.

##  organization of annotation. @Vaishnavi Jha
anonymity. Many studies are anonymised. We don't (yet) have the tools/resources to anonymize people properly but we should act as if we did. There is no sensitive data other than people's annotations but we don't want competition or blame, so we'll upload as annotator_1 ... annotator_N. Please don't try to discover identity! if we do more of this we may need to create a server.
- list of annotators.
- feedback. Please keep notes at the time of any problems, suggestions, etc.
- projected timescale.

## goals of annotation. @Ananya This is a good example of a simple tabular dataset from which we can extract more data. Examples are:
- what is the inter-annotator variance/correlation?
- which columns/concepts have the most variance?
- can we select a gold standard/ground truth (we'll need more than 30 for production)?
- do the annotation criteria need refining? (we can't answer until the first batch is analysed). should we repeat this with another 30 images? or are we ready for production?

## terminology/dictionary @Roopa.
- extract all words / phrases in images using #imageanalysis tools from @Anuv. We may need the whole 300 for this.
- look them up in Wikidata.
- create a frequency table
- create dictionary, including synonyms

## caption analysis @Shiv Kumar.

- Create dictionary of term frequency (unsupervised)
-search captions with @Roopa dictionary (supervised)
-manually classify captions with same criteria as images
- correlate caption annotations with image annotations. Can captions be used to select terpene biosynthesis pathways?

## imageanalysis 2-min update (did/will_do/blocked) @Anuv (if present) and @Peter Murray-Rust

## pygetpapers @Ayush Garg if present

## docanalysis @Shweata Hegde if present

Add other topics here.

**Comment by petermr** [comment]
# link for abstracting terpene synthase figure captions

https://github.com/wikipathways/pfocr-curation-terpene-edition

**Comment by AlexanderPico** [comment]
Just checking in. Any feedback on the curation tool? If it's too complicated or otherwise not helpful, please let me know. I won't be offended :)

I can easily provide the information on the 350 figures as a single spreadsheet (with captions text, links, etc) if that would be more useful for your classification efforts.

**Reply by Vaish2611** [reply]
Greetings Alex! 
Right Away we are taking feedbacks from the annotators for the app and are also asking them to annotate a csv file containg few figures to move further with the analysis portion. 

Soon will create a document of suggestions for the app so that we can make the changes in one go. 

Once we are done with that I will let you know the changes and if it doesn't work out will definetely ask for the spreadsheet :)

**Comment by petermr** [comment]
Thanks both!
Vaishnavi is doing a great job. She is having to create a file system for
answers (we have potentially 10 annotators)

On Fri, Jan 28, 2022 at 6:34 PM Vaishnavi ***@***.***> wrote:

> Greetings Alex!
> Right Away we are taking feedbacks from the annotators for the app and are
> also asking them to annotate a csv file containg few figures to move
> further with the analysis portion.
>
> Soon will create a document of suggestions for the app so that we can make
> the changes in one go.
>
> Once we are done with that I will let you know the changes and if it
> doesn't work out will definetely ask for the spreadsheet :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/4#discussioncomment-2068621>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS2P7R4GXFAUHK2M5FTUYLOSJANCNFSM5LMKFUJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by petermr** [comment]
# general feedback

I enjoyed doing this - I feel fairly confident that I haven't made errors such as off-by-one. Took me about 20 mins to do 30 annotations. This is because I don't think I had the right link into the pathways (I started with "Classification of 30 fig.csv" CSV file and manually paged through Alex's file https://gladstone-bioinformatics.shinyapps.io/shiny-terpeneplant/  . (His site also points to    https://gladstone-bioinformatics.shinyapps.io/shiny-25years/ which is not the correct file). 

I believe that it is possible to scroll through tis file and annotate it in one tool but I have lost the link. So I manually clicked on the next row, viewed the image, made my decision and then filled in the table, row by row. This has the danger that the right answers are filled in the wrong row. An all-in-one tool will overcome this.

Glad you are getting input!

**Reply by AlexanderPico** [reply]
Maybe I'm misunderstanding, but here is the link to the all-in-one tool: https://github.com/wikipathways/pfocr-curation-terpene-edition. This tool will let you interactively browse the set of 350 figures and record classification decisions, producing a table output.

I can easily make it work with just 30 figures, if that's more helpful.

Or, you can use the database (like you did above) and work with your own classification spreadsheet. But, I just want to make sure I didn't confuse the database link with the tool link...

**Comment by petermr** [comment]
Thanks Alex,
This is very helpful

On Tue, Feb 1, 2022 at 12:41 AM Alexander Pico ***@***.***>
wrote:

> Maybe I'm misunderstanding, but here is the link to the all-in-one tool:
> https://github.com/wikipathways/pfocr-curation-terpene-edition. This tool
> will let you interactively browse the set of 350 figures and record
> classification decisions, producing a table output.
>
That's the goal. We are learning how to run annotation projects and finding
our way

> I can easily make it work with just 30 figures, if that's more helpful.
>
I think it would be - but maybe we have done enough by hand. It will
depend on whether we want to rerun.

> Or, you can use the database (like you did above) and work with your own
> classification spreadsheet. But, I just want to make sure I didn't confuse
> the database link with the tool link...
>

No. We had a lot of discussion and it's taken work for Vaishnavi to pull it
together - she's done a good job but there is more to do.
We discussed this  in the group yesterday (and we should probably arrange a
day for you to talk with us! we normally meet at 1300 UTC which is not good
for you - we can probably rearrange a special meeting  late India, early
WestUSA).

I would like to use your tool as it's more integrated than what I did and
has fewer potential errors.
The main barrier was a reluctance to install R at this stage (I think this
will fade) - no particular problem.

The activity now is to see whether the 7-8 annotations on 30 papers give
clear signals.

The main goal is to select images which have terpene
biosynthetic pathways resulting in specific terpenes (not just general
classes)
There are 3 main axes. All cells are yes/no
* classification (4) (Pathway, Metabolism, Pre-terpene, Terpene)
* images (30) - which are useful for further study and also which show the
greatest variance;
* annotators (8)

We have enough data for Vaishnavi, Ananya and others to ask
meaningful questions about these axes:
* do any annotators deviate significantly from the main group. They are not
"wrong" but we may have to filter them,
* are any of the classifications seriously too variable to be useful
* are any of the images variable?

The majority of the group are not bioscientists so this is about whether we
can set good enough rules for classification.

In other news:
* Roopa and Anuv are working on extracting names from the diagrams
* Shiv is using NLP on the captions

We would hope to get a system where we could label a diagram (as a pathway)
by using its image content and its caption.

This is exciting.

P.

-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by Vaish2611** [comment]
Here's the CSV file to be filled. Please fill each of these columns with "yes" and "no".

[Classifiaction of 30 pathway - Sheet1 (1).csv](https://github.com/petermr/petermr/files/8117101/Classifiaction.of.30.pathway.-.Sheet1.1.csv)

Refer to these [rules](https://github.com/petermr/CEVOpen/wiki/Rules-for-Classification-of-Named-terpene-pathway) classification.
